examples/migration/recursive_fibonacci/task_emulation_layer.h

```diff
@@ -0,0 +1,145 @@
+/*
+    Copyright (c) 2023 Intel Corporation
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#ifndef __TBB_task_emulation_layer_H
+#define __TBB_task_emulation_layer_H
+
+#include "tbb/task_group.h"
+#include "tbb/task_arena.h"
+
+#include <atomic>
+
+namespace task_emulation {
+
+struct task_group_pool {
+    task_group_pool() : pool_size(std::thread::hardware_concurrency()), task_submiters(new tbb::task_group[pool_size]) {}
+
+    ~task_group_pool() {
+        for (std::size_t i = 0; i < pool_size; ++i) {
+            task_submiters[i].wait();
+        }
+
+        delete [] task_submiters;
+    }
+
+    tbb::task_group& operator[] (std::size_t idx) { return task_submiters[idx]; }
+
+    std::size_t pool_size;
```

Comment:
Perhaps, make it const so that it does not get accidentally changed?
```suggestion
    const std::size_t pool_size;
```